### PR TITLE
Allocate a TTY for running stages

### PR DESCRIPTION
**Pull request by devimc**
allocate a TTY for running stages

### CODE REVIEW

File: cmd/localCI/repo.go [hunk 1]
```diff
@@ -94,6 +94,9 @@ type Repo struct {
 
 	// logger of the repository
 	logger *logrus.Entry
+
```
Comment (grahamwhaley) [inline cmd/localCI/repo.go]: Is this patch needed as part of this series. It's not clear to me (I've not read the second patch yet), but there is nothing in the commit message or leaping out from the code at me that it is - just checking it was not a slip of the git ?
and, s/wih/with/ in the commit message

File: cmd/localCI/README.rst [hunk 2]
```diff
@@ -86,6 +86,7 @@ This section have next keys:
 - ``teardown [[]string] (optional)``: the commands executed to cleanup the environment doesn't matter if ``setup`` and ``run`` finished correctly or not.
 - ``onSuccess [[]string] (optional)``: the commands to be executed if ``setup``, ``run`` and ``teardown`` finished correctly.
 - ``onfailure [[]string] (optional)``: the commands to be executed if any of ``setup``, ``run`` or ``teardown`` fail.
+- ``tty [bool] (optional)``: specify whether **localCI** must allocate a tty to run the stages. By default this value is ``true``.
```
Comment (grahamwhaley) [inline cmd/localCI/README.rst]: nice :-)

File: cmd/localCI/console.go [hunk 3]
```diff
@@ -0,0 +1,149 @@
+// Copyright (c) 2014,2015,2016 Docker, Inc.
```
Comment (grahamwhaley) [inline cmd/localCI/console.go]: maybe you can reference where the original code came from, presuming you copied this from runc or similar? Put that comment after the copyright header probably.

File: cmd/localCI/stage_test.go [hunk 4]
```diff
@@ -31,11 +31,11 @@ func TestStageRun(t *testing.T) {
 
 	sc := stageConfig{
 		logger: logrus.WithFields(logrus.Fields{"test": "test"}),
+		tty:    false,
 	}
 
 	sc.logDir, err = ioutil.TempDir("/tmp", ".logs")
 	assert.NoError(err)
-	defer os.RemoveAll(sc.logDir)
```
Comment (grahamwhaley) [inline cmd/localCI/stage_test.go]: a sneaky extra little bug fix here? :-)

Comment (devimc) [inline cmd/localCI/stage_test.go]: nop, I remove it in the line 92

```
assert.NoError(os.RemoveAll(sc.logDir))
```

File: cmd/localCI/console.go (hunk 3 above)
Comment (devimc) [inline cmd/localCI/console.go]: I copied it from cc-runtime

File: cmd/localCI/repo.go (hunk 1 above)
Comment (devimc) [inline cmd/localCI/repo.go]: Fixed, thanks

File: cmd/localCI/console.go (hunk 3 above)
Comment (grahamwhaley) [inline cmd/localCI/console.go]: OK. I can't find it - can you point us to an URL for the one you copied, and we can try to backtrack it to attribute correctly. thx.

**Comment by devimc** [comment]
Hi @iphutch  could you review it?

**Pull request merged**